sklearn/ensemble/_hist_gradient_boosting/tests/test_warm_start.py

```diff
@@ -0,0 +1,189 @@
+import numpy as np
+import pytest
+from sklearn.base import clone
+from sklearn.datasets import make_classification, make_regression
+
+# To use this experimental feature, we need to explicitly ask for it:
+from sklearn.experimental import enable_hist_gradient_boosting  # noqa
+from sklearn.ensemble import HistGradientBoostingRegressor
+from sklearn.ensemble import HistGradientBoostingClassifier
+
+
+X_classification, y_classification = make_classification(random_state=0)
+X_regression, y_regression = make_regression(random_state=0)
+
+
+def _assert_predictor_equal(gb_1, gb_2, X):
+    """Assert that two HistGBM instances are identical."""
+    # Check identical nodes for each tree
+    for (pred_ith_1, pred_ith_2) in zip(gb_1._predictors, gb_2._predictors):
+        for (predictor_1, predictor_2) in zip(pred_ith_1, pred_ith_2):
+            np.testing.assert_array_equal(
```

Comment:
@NicolasHug Do we agreed on not using `sklearn.utils.testing.assert_*` and use the `np.testing` utils?
I don't recall but I might have missed the discussion.

Reply:
I think we don't use `assert_array_equal` from `sklearn.utils` and use these from `np.testing` now.

But I don't know. I'm not sure.
It has never been clear to me cf https://github.com/scikit-learn/scikit-learn/issues/13180.

I have decided not to care about this anymore.

Reply:
Before to merge, I will just make an explicit import

```python
from numpy.testing import assert_array_equal
```

Like this only the import will change if we are going to make it homogeneous.